src/libcore/fmt/mod.rs

```diff
@@ -166,7 +166,9 @@ pub struct Formatter<'a> {
 // NB. Argument is essentially an optimized partially applied formatting function,
 // equivalent to `exists T.(&T, fn(&T, &mut Formatter) -> Result`.
 
-enum Void {}
+struct Void {
+    _private: (),
+}
```

Comment:
What tests? I don't want to land this if it changes *semantics* of code that already compiles (as opposed to just letting more code compiles).